### PR TITLE
BZ2051452: Add L2 subnet info to vSphere, RHV, and Openstack prerequisites

### PR DESCRIPTION
**Pull request by cherraz**
Added a reusable snippet across some vSphere, RHV, and Openstack installer IPI

FIXES BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2051452

PREVIEW:
Openstack
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-installer-custom.html#installing-openstack-installer-custom#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-installer-kuryr.html#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-installer-restricted.html#prerequisites

RHV
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-default.html#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations.html#prerequisites

vSphere
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#installing-vsphere-installer-provisioned-customizations#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-network-customizations.html#prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-restricted-networks-installer-provisioned-vsphere.html#prerequisites_installing-restricted-networks-installer-provisioned-vsphere

vSphere cloud

* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-vmc.html#vsphere-prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-vmc-customizations.html#vsphere-prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-vmc-network-customizations.html#vsphere-prerequisites
* https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-restricted-networks-vmc.html#vsphere-prerequisites_installing-restricted-networks-vmc

VERSIONS: 4.10,  4.9, 4.8, 4.7, 4.6  (it doesn't apply to main)

### CODE REVIEW

File: snippets/installing-subnet-requirements.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+* If you are using L2 subnets, you ensured that either your clusters will run on dedicated subnets or that every cluster on a given subnet has a unique Virtual Router Identifier (VRID). Running clusters with the same VRID causes broadcast storms and connection reset errors.
++
+The installer process assigns VRIDs based on cluster IDs. Even if you change the cluster ID after installation, you cannot change the VRID assigned to it.
++
+To check the VRIDs, run the following command for all clusters in a given subnet, including any clusters you intend to install:
++
+[source,terminal,subs="attributes+"]
+----
+$ podman run quay.io/openshift/origin-baremetal-runtimecfg:{product-version} vr-ids <cluster_ID>
```
Comment (cherraz) [inline snippets/installing-subnet-requirements.adoc:26]: It doesn't seem very frequent to see prerequisites with commands (understandable... ), but this seemed the best place to add it.  

This command comes from here "https://access.redhat.com/solutions/6450581"  though, so another possibility would be adding a link there. But the reader will need to be logged to see it and click, so not as convenient. 

Thoughts?

Comment (cherraz) [reply snippets/installing-subnet-requirements.adoc:23]: Also I expect "{product-version} resolves to 4.7, 4.8, etc.  At the moment it resolves to "Branch Build" but I think that's fine?

**Comment by cherraz** [comment]
Hi @cybertron,  made changes to the PR as you adviced. Could you please take a look at this new one and see everything looks fine? It's basically one file with a sentence and a link. The other files are the IPI sections including this sentence.    Thank you!

**Review by cybertron — COMMENTED** [review]
Sorry, I thought I had already looked at this. A couple of thoughts inline.

File: snippets/installing-subnet-requirements.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned-network-customizations.adoc
+// * installing/installing-openstack/installing-openstack-installer-custom.adoc
+// * installing/installing-openstack/installing-openstack-installer-kuryr.adoc
+// * installing/installing-openstack/installing-openstack-installer-restricted.adoc
+// * installing/installing-rhv/installing-rhv-default.adoc
+// * installing/installing-rhv/installing-rhv-customizations.adoc
+
+:_content-type: SNIPPET
+
+* If you are using L2 subnets, you ensured that link:https://access.redhat.com/solutions/6450581[either your clusters will run on dedicated subnets or that every cluster on a given subnet has a unique Virtual Router Identifier (VRID)]. Running clusters with the same VRID in the same subnet causes broadcast storms and connection reset errors.
```
[the remark this answers is not in the document]
Comment (cherraz) [reply snippets/installing-subnet-requirements.adoc:15]: Thank you Ben! I'm not very knowledgeable about networks. I'll remove the bit about L2 subnets and will reword the broadcast storm reference.

**Comment by kelbrown20** [comment]
Looks great! LGTM

**Comment by cybertron** [comment]
/lgtm

Thanks!

**Comment by cherraz** [comment]
@zhaozhanqi  Hi!  When you can, please let me know if you approve this PR so that we can merge it. Thanks!

**Comment by jinyunma** [comment]
the update looks good to me for vsphere part.

I could not link to the docs under "Installing on VMC" to check if all IPI installation on VMC docs are updated. Those docs also need to be updated.

Another question is that can we add KCS link directly in official doc?

**Comment by cherraz** [comment]
> the update looks good to me for vsphere part.
> 
> I checked that all IPI installation on VMC docs (under ["Installing on VMC"](https://deploy-preview-43599--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vmc/installing-vmc.html)) are not applied the update. Those docs also need to be updated.
> 
> Another question is that can we add KCS link directly in official doc?

Hi! Thank you for checking this -- I'll add the note to those sections as well.  

About the link, yeah, this is not the first link to the knowledge base in the docset, so that should be ok :)

**Comment by jinyunma** [comment]
@cherraz thanks for updating. The changes look good to me now for vsphere part.

**Review by mandre — COMMENTED** [review]
The note looks good to me, however shouldn't we include it in UPI installation method as well? It also seems to me it's missing from a couple of OpenStack doc (SR-IOV and OVS-DPDK).

**Comment by cybertron** [comment]
This isn't relevant for UPI because it doesn't run keepalived. It also isn't necessary going forward because we switched everyone to unicast keepalived so it won't conflict.

**Pull request closed without merging**